### PR TITLE
 Update gitlab to vet and build golang binary (fixes BAC-209).

### PR DESCRIPTION
**Pull request by davidewatson**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -23,6 +23,22 @@ stages:
   - test
   - publish
 
+build-golang:
+  stage: build
+  image: quay.io/samsung-cnct/golang-container:latest
+  script:
+  - gometalinter.v2 --install
```
Comment (philoserf) [inline .gitlab-ci.yml]: I prefer that these common steps be in a place developers can execute locally AND the ci also call that shared source. See kraken and it's CI for an example.

Comment (guineveresaenger) [reply .gitlab-ci.yml]: The ultimate goal is a Makefile. Is that what you had in mind?

**Comment by davidewatson** [comment]
@philoserf: Just to follow up, we discussed putting the build logic in a Dockerfile/Makefile and having GitLab just call `make`. This was merged anyway so that PRs will push up an image in the meantime.

**Pull request merged**